ethcore/private-tx/src/lib.rs

```diff
@@ -511,25 +524,34 @@ impl Provider where {
 			let (new_address, _) = ethcore_contract_address(engine.create_address_scheme(env_info.number), &sender, &nonce, &transaction.data);
 			Some(new_address)
 		});
+		let contract_address = contract_address.expect("Private contract address should be non zero by this moment; qed");
+		// Patch other available private contracts' states as well
+		if let Some(key_server_account) = self.keys_provider.key_server_account() {
+			if let Some(available_contracts) = self.keys_provider.available_keys(block, &key_server_account) {
+				for private_contract in available_contracts {
```

Comment:
This may be a performance bottleneck. I don't think we need to fix it immediately, but what we can do is:

* Add a hook in `Externalities::call` that checks the contract address.
* From there, only patch an account in the state if it's actually called.

Reply:
Yes, you're right! TODO added into the code and the corresponding task #10133 created